pootle/apps/pootle_fs/plugin.py

```diff
@@ -0,0 +1,344 @@
+            self, fs_path=fs_path, pootle_path=pootle_path)
+
+    @responds_to_state
+    def add(self, state, response,
+            fs_path=None, pootle_path=None, force=False):
+        """
+        Add translations from Pootle into the FS
+
+        If ``force``=``True`` is present it will also:
+        - add untracked conflicting files from Pootle
+        - mark tracked conflicting files to update from Pootle
+
+        :param force: Add conflicting translations.
+        :param fs_path: Path glob to filter translations matching FS path
+        :param pootle_path: Path glob to filter translations to add matching
+          ``pootle_path``
+        """
+        to_add = state["pootle_untracked"]
+        if force:
+            to_add = (
+                to_add
+                + state["conflict_untracked"]
+                + state["fs_removed"]
+                + state["conflict"])
+        for fs_state in to_add:
+            if fs_state.state_type in ["pootle_untracked", "conflict_untracked"]:
+                fs_state.kwargs["store_fs"] = self.create_store_fs(
+                    pootle_path=fs_state.pootle_path,
+                    fs_path=fs_state.fs_path)
+            fs_state.store_fs.file.add()
+            response.add("added_from_pootle", fs_state=fs_state)
+        return response
+
+    @responds_to_state
+    def fetch(self, state, response,
+              fs_path=None, pootle_path=None, force=False):
+        """
+        Add translations from FS into Pootle
+
+        If ``force``=``True`` is present it will also:
+        - add untracked conflicting files from FS
+        - mark tracked conflicting files to update from FS
+
+        :param force: Add conflicting translations.
+        :param fs_path: Path glob to filter translations matching FS path
+        :param pootle_path: Path glob to filter translations to add matching
+          ``pootle_path``
+        """
+        to_fetch = state["fs_untracked"]
+        if force:
+            to_fetch = (
+                to_fetch
+                + state["conflict_untracked"]
+                + state["pootle_removed"]
+                + state["conflict"])
+        for fs_state in to_fetch:
+            if fs_state.state_type in ["fs_untracked", "conflict_untracked"]:
+                fs_state.kwargs["store_fs"] = self.create_store_fs(
+                    pootle_path=fs_state.pootle_path,
+                    fs_path=fs_state.fs_path)
+            fs_state.store_fs.file.fetch()
+            response.add("fetched_from_fs", fs_state=fs_state)
+        return response
+
+    @responds_to_state
+    def merge(self, state, response,
+              fs_path=None, pootle_path=None, pootle_wins=False):
+        """
+        :param fs_path: Path glob to filter translations matching FS path
+        :param pootle_path: Path glob to filter translations to add matching
+          ``pootle_path``
+        """
+        for fs_state in state["conflict"] + state["conflict_untracked"]:
```

Comment:
I suggested this to align with the other methods, so code follows the same pattern.